lib/api/impl/patient_api_impl.dart

```diff
@@ -60,4 +60,50 @@ class PatientApiImpl extends PatientApi {
   Future<List<String>?> matchPatients(Filter filter) {
     return _api.basePatientApi.rawMatchPatientsBy(filter.toAbstractFilterDto());
   }
+
+  @override
+  Future<Patient> giveAccessTo(Patient patient, String delegatedTo) async {
+    final localCrypto = _api.crypto;
+    final currentUser = await _api.baseUserApi.getCurrentUser();
+
+    // Check if delegatedBy has access
+    if (!patient.systemMetaData!.delegations.entries.any((element) => element.key == currentUser!.dataOwnerId())) {
+      throw StateError("Couldn't give access to unowned dataSample");
```

Comment:
StateError("DataOwner $currentUser.dataOwner() does not have the right to access patient $patient.id")